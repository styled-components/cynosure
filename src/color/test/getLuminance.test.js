// @flow
import getLuminance from '../getLuminance'

describe('getLuminance', () => {
  it('should return the luminance of a hex color', () => {
    expect(getLuminance('#444')).toEqual(0.058)
  })

  it('should return the luminance of an 8-digit hex color', () => {
    expect(getLuminance('#6564CDB3')).toEqual(0.163)
  })

  it('should return the luminance of an 4-digit hex color', () => {
    expect(getLuminance('#0f08')).toEqual(0.715)
  })

  it('should return the luminance of an rgba color', () => {
    expect(getLuminance('rgba(101,100,205,0.7)')).toEqual(0.163)
  })

  it('should return the luminance of an rgb color', () => {
    expect(getLuminance('rgb(204,205,100)')).toEqual(0.574)
  })

  it('should return the luminance of an hlsa color', () => {
    expect(getLuminance('hsla(250, 100%, 50%, 0.2)')).toEqual(0.077)
  })

  it('should return the luminance of an hls color', () => {
    expect(getLuminance('hsl(0, 100%, 50%)')).toEqual(0.213)
  })

  it('should return the luminance of a named CSS color', () => {
    expect(getLuminance('papayawhip')).toEqual(0.878)
  })

  it('should return 0 when passed transparent', () => {
    expect(getLuminance('transparent')).toEqual(0)
  })
})
