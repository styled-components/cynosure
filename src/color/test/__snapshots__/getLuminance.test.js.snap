// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getLuminance should return the luminance of a hex color 1`] = `0.05780543019106723`;

exports[`getLuminance should return the luminance of a named CSS color 1`] = `0.877971001998354`;

exports[`getLuminance should return the luminance of an 8-digit hex color 1`] = `0.16288822420427432`;

exports[`getLuminance should return the luminance of an hls color 1`] = `0.2126`;

exports[`getLuminance should return the luminance of an hlsa color 1`] = `0.07733591265855211`;

exports[`getLuminance should return the luminance of an rgb color 1`] = `0.5742011250098834`;

exports[`getLuminance should return the luminance of an rgba color 1`] = `0.16288822420427432`;
